src/coreclr/jit/helperexpansion.cpp

```diff
@@ -1060,6 +1068,14 @@ bool Compiler::fgExpandThreadLocalAccessForCall(BasicBlock** pBlock, Statement*
         // Create tree to "threadStaticBlockValue = threadStaticBlockBase[typeIndex]"
         typeThreadStaticBlockIndexValue = gtNewOperNode(GT_MUL, TYP_INT, gtCloneExpr(typeThreadStaticBlockIndexValue),
                                                         gtNewIconNode(TARGET_POINTER_SIZE, TYP_INT));
+        // Usually a constant, try to fold
+        typeThreadStaticBlockIndexValue = gtFoldExpr(typeThreadStaticBlockIndexValue);
```

Comment:
should this be or do we need to fold expression before casting it as well?

```c++
#ifdef TARGET_X64
    typeThreadStaticBlockIndexValue = gtNewCastNode...
#endif
    typeThreadStaticBlockIndexValue = gtFoldExpr(...
```

Reply:
We need to fold it before as well since `gtFoldExpr` is not recursive.